message_row: Place icon representations of edited, moved messages.

karlstolley opened the pull request:
This PR uses icons to indicate EDITED and MOVED states on messages.

[CZO discussion](https://chat.zulip.org/#narrow/channel/101-design/topic/icons.20for.20edited.20and.20moved.20markers/near/2157017)

As of 2025-07-02, the moved icon is shown on hover, when a message is selected, or when in a touchscreen environment:

![show-moved-icon-on-selected](https://github.com/user-attachments/assets/b0badff3-f5c3-4048-8da8-9984f6802933)

As of 2025-06-16, the PR uses different icons as suggested by @terpimost. Additionally, in the saving-the-edit state, the bouncing vdots device is shown.

As of 2025-06-10, the PR increases the space between the edited/moved icon and the sender name. It also introduces a pulse effect on the edit icon, in place of the previous flashing of SAVING text.

| Sender-line spacing, before | After |
| --- | --- |
| ![sender-line-icon-align-before](https://github.com/user-attachments/assets/a3c58db4-a1e6-4b25-a7a4-6c0338518257) | ![sender-line-icon-align-after](https://github.com/user-attachments/assets/1b1ebc06-9fe4-443c-a184-60d2c9da6576) |

| Pulsing save state |
| --- |
| ![pulse-edit-icon-on-save](https://github.com/user-attachments/assets/2ea38e0b-be14-4714-80dc-0d5226098df8) |

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

| Before | After, showing edited & moved | After, showing edited only |
| --- | --- | --- |
| ![edited-notices-before](https://github.com/user-attachments/assets/ce1fa633-0a1a-4f77-8158-951d72a5662a) | ![edited-icons-both-after](https://github.com/user-attachments/assets/b9650381-6e8e-49ac-999b-f40d37a61d31) | ![edit-icons-edited-only-after](https://github.com/user-attachments/assets/a528cc06-ed75-4b65-b26a-db5458e49db5) |

| Showing moved on hover |
| --- |
| ![show-moved-icons-on-hover](https://github.com/user-attachments/assets/f4f565f7-5fcb-4937-9ebc-a96e08e98a38) |

karlstolley committed:
message_row: Conditionally display moved icons.

alya commented:
Hm, did you test clicking on these icons? For me, it only works if I click in the very upper left corner.

alya commented:
With he edit icon, I'm seeing a pretty significant visual shift between the live echo and the final version. Can that be avoided?

alya commented:
In the gutter area, the three dots aren't really in the same place as the pencil at the moment. They are way over in the upper left.

Looks good in manual testing other than that!

karlstolley committed:
message_row: Display 3-dots animation on edit save.

This commit also squares the display and layout of
.edit-notifications and .message_edit_notice, so that
the vdots animation looks identical, regardless of
whether they are displayed while a user awaits their
edit to save, or a second user awaits the editing user
to finish editing.

karlstolley committed:
message_row: Display icons for edited, moved.

amanagr commented:
Please add the czo review label back when it is ready for a deploy.
